### ISSUE TITLE
TCP/IP stack sometimes counts output incorrectly on some devices

### ISSUE THREAD

**Issue by sfionov** [AdguardTeam/CoreLibs#153]
May be an NDK bug - incorrect handling of sendmsg() returning `-1` due to code optimization.

```20:45:51.596 [VpnService-main] DEBUG Native.TCPIP.UDPMNGR - [LWIP id=7] new UDP connection request 172.16.209.1:51875 -> 8.8.8.8:53
20:45:51.597 [pool-218-thread-1] DEBUG c.a.a.f.vpn.FilteringVpnService - UDP id=7 onUdpConnectRequest: protocol=UDP, local=/172.16.209.1:51875, remote=/8.8.8.8:53
20:45:51.599 [VpnService-main] DEBUG Native.TCPIP.UDPMNGR - [LWIP id=7] complete_udp_connect_request: forwarding connection
20:45:51.601 [pool-218-thread-1] DEBUG c.a.a.f.vpn.FilteringVpnService - UDP id=7 Connection statistic: sent=4308329427 received=0
20:45:51.602 [pool-1-thread-1] DEBUG com.adguard.android.service.as - Increment statistics cache
20:45:51.602 [pool-1-thread-1] DEBUG com.adguard.android.service.as - Increment statistics cache
20:45:51.830 [main] INFO  c.a.a.r.NetworkStateReceiver - Start NetworkStateReceiver on action android.net.conn.CONNECTIVITY_CHANGE
```

**Comment by sfionov** [comment]
Adguard for Android version: 2.10.176
Android version: 6.0.1
Device model: ZTE Nubia Z17 Mini

**Comment by sfionov** [comment]
Fixed in CL 1.0.25
